### PR TITLE
pass initial kafka timestamp from Ender -> Vulcan for better e2e order latency tracking

### PR DESCRIPTION
**Pull request by dydxwill**
### Changelist
pass initial kafka timestamp from Ender -> Vulcan for better e2e order latency tracking

### Test Plan
unit tested.

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: indexer/services/ender/src/handlers/handler.ts [hunk 1]
```diff
@@ -49,20 +51,23 @@ export abstract class Handler<T> {
...
     event: T,
+    messageReceivedTimestamp?: string,
   ) {
     this.block = block;
     this.blockEventIndex = blockEventIndex;
     this.indexerTendermintEvent = indexerTendermintEvent;
     this.timestamp = DateTime.fromJSDate(block.time!);
     this.txId = txId;
     this.event = event;
+    this.messageReceivedTimestamp = messageReceivedTimestamp;
```
Comment (Christopher-Li) [inline indexer/services/ender/src/handlers/handler.ts:70]: can't we just use the timestamp from `this.block`? I'm pretty sure we already have a helper to pull timestamp out of it

File: indexer/services/ender/src/lib/block-processor.ts [hunk 2]
```diff
@@ -77,13 +77,16 @@ export class BlockProcessor {
...
+  messageReceivedTimestamp: string;
 
   constructor(
     block: IndexerTendermintBlock,
     txId: number,
+    messageReceivedTimestamp: string,
   ) {
     this.block = block;
     this.txId = txId;
+    this.messageReceivedTimestamp = messageReceivedTimestamp;
```
Comment (Christopher-Li) [inline indexer/services/ender/src/lib/block-processor.ts:89]: same as above

**Review by Christopher-Li — COMMENTED** [review]
Are you going to add metrics to Vulcan in another PR?

**Comment by dydxwill** [comment]
> Are you going to add metrics to Vulcan in another PR?

yeah

File: indexer/services/ender/src/handlers/handler.ts (hunk 1 above)
Comment (dydxwill) [reply indexer/services/ender/src/handlers/handler.ts:70]: We could, but I wanted to only track Indexer system latency without block gossiping latency.

**Pull request merged**